benchmarks/timeseries/resampling.py

```diff
@@ -9,13 +9,13 @@
 
 from frequenz.quantities import Quantity
 
-from frequenz.sdk.timeseries import ResamplerConfig, Sample
+from frequenz.sdk.timeseries import ResamplerConfig
 from frequenz.sdk.timeseries._resampling._base_types import SourceProperties
 from frequenz.sdk.timeseries._resampling._resampler import _ResamplingHelper
 
 
 def nop(  # pylint: disable=unused-argument
-    samples: Sequence[Sample[Quantity]],
+    samples: Sequence[tuple[datetime, Quantity]],
```

Comment:
I wonder if a `NoNoneSample` type or so might be a nice abstraction here. I can imagine there are actually many places that would enjoy not having to check for `None` after they initially received the data. I can certainly use it in FCR as well.

For now its fine.

Reply:
Yeah, in any case here we want to get rid of even the `Quantity`, we shouldn't really use `Sample` either, at least until we can use `Sample[float]`.

If we go with the `None`/not `None` sample, maybe it can be `OptionalSample` (`None` values allowed) and `Sample` (can't have a `None` value) to follow Python's old `Optional` as an alias of `| None`. Another option, which I'm not sure if it is doable or not, is `Sample[Power]` vs `Sample[Power | None]`. I have the feeling we tried to make that work and failed, but maybe I'm just remembering wrongly.